tests/DeploymentPipelineStepTest.php

```diff
@@ -102,6 +103,21 @@ public function testSnapshotSkipped() {
 	}
 
 	/**
+	 * Test snapshot is skipped upon user request.
+	 */
+	public function testSnapshotSkippedByUser() {
+		$step = $this->getDummyDeployment(false, true);
+		$step->start();
+
+		// Assert not error at startup
+		$this->assertEquals('Started', $step->Status);
+		$this->assertEquals('Deployment', $step->Doing);
+		$this->assertFalse(PipelineTest_MockLog::has_message('TestDeployStep:Snapshot creating snapshot of database'));
+		$snapshot = $step->Pipeline()->PreviousSnapshot();
```

Comment:
Can you assert that a deployment was started as well? Since that is the expected behaviour?